shared/constants/types/fs.js

```diff
@@ -166,20 +206,22 @@ export const getLocalPathName = (p: LocalPath): string => p.split(localSep).pop(
 export const getLocalPathDir = (p: LocalPath): string => p.slice(0, p.lastIndexOf(localSep))
 
 type PathItemComparer = (a: PathItem, b: PathItem) => number
+type PathItemLessThan = (a: PathItem, b: PathItem) => boolean
+
+const _comparerFromLessThan = (lt: PathItemLessThan): PathItemComparer => (a, b) =>
+  lt(a, b) ? -1 : lt(b, a) ? 1 : 0
```

Comment:
i'm not so worried about perf but you normally don't see this as you don't normally have to do this comparison twice. usually the 0 case is trivial to calc and then you run the compare once to decide 1: -1

Reply:
The existing implementations did exactly this...figured it was better to have cleaner code here.

Reply:
FWIW this is Go's API for `sort.Sort`: you only need to implement `Len`, `Less`, and `Swap`.